state/factory/workingset.go

```diff
@@ -156,10 +161,7 @@ func (ws *workingSet) runAction(
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, errors.Wrap(action.ErrHitGasLimit, "block gas limit exceeded")
 	}
-	// Reject execution of chainID not equal the node's chainID
-	if elp.ChainID() != config.ChainID() {
```

Comment:
should check here:
`if elp.ChainID() != blkChainCtx.ChainID`

Reply:
elp.ChainID() from client side: client wants to target to which chain
blkChainCtx.ChainID from server side: it is the chainID that was used to start the node